posts/2025-12-04-project-goals-nov-update.md

```diff
@@ -0,0 +1,240 @@
+* rust-lang/rust#125116 may be contested in results. The impact may not be as large as expected, even on Clippy.
+* We've been experimenting with Clippy using `rustc_driver` as a static library, instead of dynamic linking. This would be us both a way to check the performance impact of `rustc_driver` as a shared library, **and** a way to profile Clippy without filtering between `dl_*` calls.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/115'><strong>Patterns of empty types</strong></a></div>
+    <div style="flex: initial;"><progress value="3" max="8"></progress>
+</div>
+</div>
+
+* The [never patterns RFC](https://github.com/rust-lang/rfcs/pull/3719) was posted.
+* Feedback on the RFC suggests that the question of "which arms can be omitted" isn't as orthogonal as hoped, so the focus will switch to that.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/101'><strong>Provided reasons for yanked crates</strong></a></div>
+    <div style="flex: initial;"><progress value="2" max="9"></progress>
+</div>
+</div>
+
+* The PR https://github.com/rust-lang/crates.io/pull/9423 has been merged.
+* Work is ongoing on the frontend feature.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/118'><strong>Scalable Polonius support on nightly</strong></a></div>
+    <div style="flex: initial;"><progress value="1" max="3"></progress>
+</div>
+</div>
+
+* Amanda's EuroRust talk on polonius from last month is also now [available on YouTube](https://www.youtube.com/watch?v=uCN_LRcswts&feature=youtu.be).
+* Implementation work continues, mostly on a branch. Major developments include a new debugger which has accelerated progress. There are about 70 test failures left to be analyzed.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/119'><strong>Stabilize cargo-script</strong></a></div>
+    <div style="flex: initial;"><progress value="27" max="35"></progress>
+</div>
+</div>
+
+* rust-lang/cargo#14670 and rust-lang/cargo#14749 have been posted and merged.
+* rust-lang/cargo#14792 has been posted.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/121'><strong>Stabilize parallel front end</strong></a></div>
+    <div style="flex: initial;"><progress value="1" max="3"></progress>
+</div>
+</div>
+
+* Still in the process of determining the cause of the deadlock through local testing and compiler code analysis.
+* **Help wanted:** Help test the deadlock code in the [issue list](https://github.com/rust-lang/rust/labels/WG-compiler-parallel) and try to reproduce the issue.
```

Comment:
Maybe this should be plural, there seem to be quite a few reported deadlocks unfortunately.

Reply:
```suggestion
* Still in the process of determining the cause of the deadlock through local testing and compiler code analysis.
* **Help wanted:** Try to reproduce deadlocks described in the [issue list](https://github.com/rust-lang/rust/labels/WG-compiler-parallel).
```